src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultString = '';
+
+    if (!this.length) {
+      return resultString;
+    }
+
+    for (const element of this) {
+      if (element !== null && element !== undefined) {
```

Comment:
The original join function also excludes empty elements (undefined, null, and empty strings). You need to add a check for empty strings too.